### DISCUSSION TITLE
Can't create API Client unless URL is provided (or when a URL is provided but requires a token)

### DISCUSSION THREAD

**Discussion by cpbr-dev** [scalar/scalar#2555] [Q&A]
Currently running into a bit of an issue with what I'm working on. The spec for the API I'm documenting is only available through an endpoint of that API (/definition), but the API requires a Bearer Token to access it for security reasons. Is there no way to pass this token to Scalar so it can fetch the spec from the URL ? 

> [@scalar/api-client-modal] Could not create the API client. Please provide an OpenAPI document: { spec: { url: '…' } } Read more: https://github.com/scalar/scalar/tree/main/packages/api-client-modal

There's also a slight issue in the error, the URL it shows no longer exists.

**Comment by hanspagel** [comment]
Submitted a PR for the broken links: #2556

**Comment by hanspagel** [comment]
Submitted a PR to just create an empty workspace if no OpenAPI document is passed: #2557 :)

**Comment by hanspagel** [comment]
There is an `updateAuth` method returned by `createClientAppModal`, does that help?

**Reply by cpbr-dev** [reply]
Thanks for the reply, could you please elaborate on this ?
To give more context, I'm running Scalar from a basic HTML file [following this](https://github.com/scalar/scalar/blob/main/documentation/integrations/html.md). How could I access the method with plain old JS in this case ? I must admit I'm still in the learning stages when it comes to this.
